config.json

```diff
@@ -18,6 +18,17 @@
 	    ]
 	},
 	"propagatedmount": "/mnt",
+	"mounts": [
+		{
+		"name": "KeyPath",
+		"description": "Path to SSH keys",
+		"source": "/tmp",
+		"destination": "/tmp/sshfs",
```

Comment:
@stevvooe I'm not entirely sure if I understand the question, so please bear with me.  By "container" in "...parameterizing for each volume, rather than container,...", do you mean the possible plugin containers?  (e.g. the runc containers that are running the plugin code?)  Or do you mean each container that requests a volume?

If the first, AFAIK (@davidwilliamson and @vieux please correct me if I'm wrong) only one instance of the plugin container runs on each node?  So the multiple SSHFS volumes that can be created will all be mounted into that one container, and I guess provided to other containers that call it? (after that I have no idea how it gets attached to running containers that require the volume).

If you mean mapping container that requests an sshfs volume to a volume that contains keys, that can't be done prior to the plugin install.  I think the volume plugin also doesn't know anything about the containers that request it, other than the [caller ID](https://docs.docker.com/engine/extend/plugins_volume/#/volumedrivermount) (is this the container ID maybe?).  

Even if we added the ssh identities via secrets though, since there is just one global plugin, I think all users would have access to all the ssh identities mounted in via secrets.  

That said only other option I can think of, to implement what I think you are suggesting (have the user mount the ssh key or otherwise provide the SSH key into the container that desires the sshfs volume, and have the volume driver use that) is to give the plugin permission to mount the docker socket and introspect containers (this is assuming the caller ID on the mount API is the container ID), and somehow exec into that or otherwise copy the identity out of that container?